src/uct/ib/base/ib_md.c

```diff
@@ -626,9 +627,39 @@ ucs_status_t uct_ib_memh_alloc(uct_ib_md_t *md, size_t length,
     return UCS_OK;
 }
 
-uint64_t uct_ib_memh_access_flags(uct_ib_mem_t *memh, int relaxed_order)
+uint64_t uct_flags_to_ibv_mem_access_flags(uint64_t uct_flags)
 {
-    uint64_t access_flags = UCT_IB_MEM_ACCESS_FLAGS;
+    uint64_t ibv_flags = 0;
+
+    if (uct_flags & UCT_MD_MEM_ACCESS_LOCAL_WRITE) {
+        ibv_flags |= IBV_ACCESS_LOCAL_WRITE;
+    }
+
+    if (uct_flags & UCT_MD_MEM_ACCESS_REMOTE_GET) {
+        ibv_flags |= IBV_ACCESS_REMOTE_READ;
+    }
+
+    if (uct_flags & UCT_MD_MEM_ACCESS_REMOTE_PUT) {
+        ibv_flags |= IBV_ACCESS_REMOTE_WRITE;
+    }
+
+    if (uct_flags & UCT_MD_MEM_ACCESS_REMOTE_ATOMIC) {
+        ibv_flags |= IBV_ACCESS_REMOTE_ATOMIC;
+    }
+
+    return ibv_flags;
+}
+
+uint64_t uct_ib_memh_access_flags(uct_ib_mem_t *memh, int relaxed_order,
+                                  int first_attempt, uint64_t uct_flags)
```

Comment:
Here as well, I'd rename `first_attempt` to its real meaning in this context.
Maybe smth like `use_uct_flags`?